src/compiler/resolutionCache.ts

```diff
@@ -205,6 +206,30 @@ export function removeIgnoredPath(path: Path): Path | undefined {
         path;
 }
 
+function perceivedOsRootLengthForWatching(pathComponents: Readonly<PathPathComponents>, length: number) {

```

Comment:
This logic is same as before except it returns number of components that are roots for watching. Like `c:\users\someUserName` or in case of linux `/anything/anything` etc. So we would never watch anything in those directories.